python/ray/train/_internal/schema.py

```diff
@@ -0,0 +1,63 @@
+from typing import List, Optional
+
+from ray._private.pydantic_compat import BaseModel, Field
+from ray.util.annotations import DeveloperAPI
+
+try:
+    import pydantic  # noqa: F401
+except ImportError:
+    raise ModuleNotFoundError(
+        "pydantic isn't installed. "
+        "To install pydantic, please run 'pip install pydantic'"
+    )
+
+
+@DeveloperAPI
+class TrainWorkerInfo(BaseModel):
+    """Metadata of a Ray Train worker."""
+
+    actor_id: str = Field(description="Actor ID of the worker.")
+    world_rank: int = Field(description="World rank.")
+    local_rank: int = Field(description="Local rank.")
+    node_rank: int = Field(description="Node rank.")
+    gpu_ids: Optional[List[str]] = Field(
+        description="A list of GPU ids allocated to that worker."
+    )
+    node_id: Optional[str] = Field(
+        description="ID of the node that the worker is running on."
+    )
+    node_ip: Optional[str] = Field(
+        description="IP address of the node that the worker is running on."
+    )
+    pid: Optional[str] = Field(description="PID of the worker.")
+
+
+@DeveloperAPI
+class TrainDatasetInfo(BaseModel):
+    name: str = Field(
+        description="The key of the dataset dict specified in Ray Train Trainer."
+    )
+    plan_name: str = Field(description="The name of the internal dataset plan.")
+    plan_uuid: str = Field(description="The uuid of the internal dataset plan.")
```

Comment:
Why do we need the plan name if we have the UUID?

Reply:
Ray Data team recommends using `dataset_tag = f"{plan_name}_{plan_uuid}` as a unique id for a dataset. 
https://github.com/ray-project/ray/blob/98e51b5aa310c4be1880c8fe739b6b8f8ff541e3/python/ray/data/_internal/plan.py#L510

Also, from observability team, the `dataset_tag` is they primary key they use in Dashboard.

Reply:
It also makes it easy to support stream_split_op in the future.

https://github.com/ray-project/ray/blob/98e51b5aa310c4be1880c8fe739b6b8f8ff541e3/python/ray/data/_internal/iterator/stream_split_iterator.py#L119

Reply:
Should we ask Ray Data team for a better way to get the tag?